javaslang/src/main/java/javaslang/Value.java

```diff
@@ -508,6 +521,15 @@ default CharSeq toCharSeq() {
     }
 
     /**
+     * Converts this to a {@link CompletableFuture}
+     *
+     * @return A new {@link CompletableFuture} containing the value
+     */
+     default <U extends T> CompletableFuture<T> toCompletableFuture() {
+         return CompletableFuture.completedFuture(Boolean.TRUE).thenApply(t -> get());
```

Comment:
This is a good way to handle exceptions in the CompletionStage. However, as you mentioned below, it makes sense to handle Try.NonFatalException in a special way.

Because FatalException and NonFatalException cannot be created outside of a Try, they are under our control and it is safe to unbox them. I.e. we know that their only purpose is to wrap exceptions in a RuntimeException.

From what I see the code should look look this:

```java
final CompletableFuture<T> completableFuture = new CompletableFuture<>();
try {
    completableFuture.complete(get());
} catch(Try.FatalException x) {
    throw x;
} catch(Try.NonFatalException x) {
    completableFuture.completeExceptionally(x.getCause());
} catch(Throwable x) {
    completableFuture.completeExceptionally(x);
}
return completableFuture;
```